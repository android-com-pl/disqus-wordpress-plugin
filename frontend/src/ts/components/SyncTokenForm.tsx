import * as React from 'react';
import SyncStatus from '../reducers/SyncStatus';
import { IFormProps } from './FormProps';

const SyncTokenForm = (props: IFormProps) => {
    return (
        <form name='site' action='' method='POST' onSubmit={props.onSubmitSiteForm} autoComplete='off'>
            <table className='form-table'>
                <tbody>
                    <tr>
                        <th scope='row'>
                            <label htmlFor='disqus_sync_token'>
                                {__('Site Secret Key')}
                            </label>
                        </th>
                        <td>
                            <input
                                type='text'
                                id='disqus_sync_token'
                                name='disqus_sync_token'
                                className='regular-text'
                                value={props.data.localAdminOptions.get('disqus_sync_token') || ''}
                                onChange={props.onInputChange.bind(null, 'disqus_sync_token')}
                            />
                            <p className='description'>
                                {__('Enter any combination of letters and numbers, at least 32 characters.')}
                                <br />
                                <a href='#' onClick={props.onGenerateRandomSyncToken}>{__('Generate a key')}</a>
                            </p>
                        </td>
                    </tr>
                </tbody>
            </table>
            <p className='submit'>
                <input
                    type='submit'
                    name='submit-site-form'
                    className='button button-primary'
                    value={__('Save')}
                />
            </p>
        </form>
    );
};

export default SyncTokenForm;
