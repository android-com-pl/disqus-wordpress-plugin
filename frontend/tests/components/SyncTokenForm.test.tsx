import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import SyncTokenForm from '../../src/ts/components/SyncTokenForm';
import { getDefaultFormProps } from '../fixtures';

test('SupportLinks renders with correct links', () => {
    const component = TestRenderer.create(<SyncTokenForm {...getDefaultFormProps()} />);

    expect(component).toMatchSnapshot();
});
