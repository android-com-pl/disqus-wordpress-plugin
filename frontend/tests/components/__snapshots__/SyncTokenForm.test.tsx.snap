// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncTokenForm renders with correct snapshot 1`] = `
<form
  action=""
  autoComplete="off"
  method="POST"
  name="site"
  onSubmit={[MockFunction]}
>
  <table
    className="form-table"
  >
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <label
            htmlFor="disqus_sync_token"
          >
            Site Secret Key
          </label>
        </th>
        <td>
          <input
            className="regular-text"
            id="disqus_sync_token"
            name="disqus_sync_token"
            onChange={[Function]}
            type="text"
            value=""
          />
          <p
            className="description"
          >
            Enter any combination of letters and numbers, at least 32 characters.
            <br />
            <a
              href="#"
              onClick={[MockFunction]}
            >
              Generate a key
            </a>
          </p>
        </td>
      </tr>
    </tbody>
  </table>
  <p
    className="submit"
  >
    <input
      className="button button-primary"
      name="submit-site-form"
      type="submit"
      value="Save"
    />
  </p>
</form>
`;
